src/test/java/server/HTTPServerTest.java

```diff
@@ -3,45 +3,74 @@
 import org.junit.jupiter.api.Test;
 
 import java.io.IOException;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.Map;
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 public class HTTPServerTest {
 
     @Test
     void itReturns200OKForExistingResource() throws IOException {
-        String testRequest = "GET / HTTP/1.1\r\nContent-Length:0\r\n";
+        String testRequestString = "GET / HTTP/1.1\r\nContent-Length:0\r\n";
         TestReaderWriter testReaderWriter = new TestReaderWriter()
-                .send(testRequest);
+                .send(testRequestString);
 
-        new HTTPServer().call(testReaderWriter);
+        Request testRequest = new Request(
+                "GET",
+                "/",
+                Map.of(
+                        "Content-Length",
+                        "0")
+        );
+        RequestParsable testRequestParser = new TestRequestParser(testRequest);
+
+        new HTTPServer(testRequestParser).call(testReaderWriter);
 
         assertEquals(List.of(
                 "HTTP/1.1 200 OK\r\nContent-Length:0\r\n\n"
         ), testReaderWriter.received());
     }
 
     @Test
-    void itReturns200OKForSimpleGet() throws IOException {
-        String testRequest = "GET /simple_get HTTP/1.1\r\nContent-Length:0\r\n";
+    void itRespondsToSimpleGet() throws IOException {
+        String testRequestString = "GET /simple_get HTTP/1.1\r\nContent-Length:0\r\n";
```

Comment:
**suggestion (non-blocking)**: Consider refactoring to use `send` to better represent multiple lines of the HTTP request.

```
new TestReaderWriter()
  .send("GET / HTTP/1.1\r\n")
  .send("Content-Length": "0")
  .send("\r\n");
```